docs: consul on k8s doesn't support external servers requiring mTLS (#8484)
hashicorp/consul: website/pages/docs/k8s/installation/deployment-configurations/servers-outside-kubernetes.mdx

## Configuring TLS with Auto-encrypt

-> **Note:** Consul on Kubernetes currently does not support external servers that require mutual authentication
for the HTTPS clients of the Consul servers, that is when servers have either
`verify_incoming` or `verify_incoming_https` set to `true`.
As noted in the [Security Model](docs/internals/security#secure-configuration),

> that setting isn't strictly necessary to support Consul's threat model since we recommend that
> all requests must contain a valid ACL token.

Reviewer: ```suggestion
that setting isn't strictly necessary to support Consul's threat model as it is recommended that
all requests contain a valid ACL token.
```
